libraries/AP_Networking/AP_Networking_PPP.cpp

```diff
@@ -169,28 +174,33 @@ void AP_Networking_PPP::ppp_loop(void)
     while (!hal.scheduler->is_system_initialized()) {
         hal.scheduler->delay_microseconds(1000);
     }
-    
-    // connect and set as default route
-    LWIP_TCPIP_LOCK();
-    ppp_connect(ppp, 0);
-
-    netif_set_default(pppif);
-    LWIP_TCPIP_UNLOCK();
 
     // ensure this thread owns the uart
     uart->begin(0);
     uart->set_unbuffered_writes(true);
 
     while (true) {
-        uint8_t buf[1024];
-        auto n = uart->read(buf, sizeof(buf));
-        if (n > 0) {
-            LWIP_TCPIP_LOCK();
-            pppos_input(ppp, buf, n);
-            LWIP_TCPIP_UNLOCK();
-        } else {
-            hal.scheduler->delay_microseconds(200);
+        // connect and set as default route
+        LWIP_TCPIP_LOCK();
+        ppp_connect(ppp, 0);
+
+        netif_set_default(pppif);
+        LWIP_TCPIP_UNLOCK();
+
+        need_restart = false;
+
+        while (!need_restart) {
+            uint8_t buf[1024];
```

Comment:
Curious, does moving this outside of the loop make any difference?

Reply:
not really, it's a stack allocation, doesn't matter a lot where we move the stack pointer